### PR TITLE
[BAEL-4381] Introduction to ArchUnit

### PR DESCRIPTION
**Pull request by psevestre**
Article link: http://inprogress.baeldung.com/wp-admin/post.php?post=220885&action=edit&classic-editor
Jira: http://jira.baeldung.com/browse/BAEL-4381

### CODE REVIEW

File: libraries-testing/article-resources/archunit/figure1.drawio [hunk 1]
```diff
@@ -0,0 +1 @@
.../83.0.4103.122 Electron/9.1.1 Safari/537.36" etag="taZU8w0iOhZ7shLjO8K-" version="13.5.1" type="device"><diagram name="Page-1" id="b5b7bab2-c9e2-2cf4-8b2a-24fd1a2a6d21">7VjbctowEP0aP9LBNgb3sSa3BzJNJ+2kk5eMsBdbrax1ZUEgX1/Jlm84YWgbKL3wwFhHa3l99pyVwHKn6fpSkCy5xgiY5QyjteWeWY7jj331rYFNCXjOsARiQaMSshvglj6BAauwJY0g7wRKRCZp1gVD5BxC2cGIEPjYDVsg6z41IzH0gNuQsD56RyOZGNQev20mroDGiXm070zKiTkJv8YCl9w8jyOHciYl1TLmHfOERPjYgtxzy50KRFlepespMM1qxVh538ULs3XKArjc54YrsrkffLj+Bp8ePr+/GTw409XdwKyyImxpqEg3JMsYDYmkyE3mclMRpV4i05eaYRCWGyyQy1szb6txnpGQ8vgjZhoYKkSSecXpyIxrKu1RCdxgTovnuWcMFmomSGTKzJIrEFLlw94xGusIqdcOiBmZeJ3HBUkp0wqcLUMaEZX5FHmOOreAkTmwoK7WFBmK4pXcRfHRS1DGWvhF8VF4n+iKNZUYrFuQIf4SMAUpNiqkmvWMCIw/7EoUj43aJr7BkpbQRq4BiVF4XK/dlFpdmGr/QOUnvcpbzpikmlomi7dujyLIgEd5DcadEDPa0gpEymNmiEImGCMn7LxBg6IUoLPUwmhiZljKR4FfQMqNaRhkKbErjYjkSXG//ToSeLHUOS5FCDvodE3XIiIGuSPOK+M0NzuFI4ApB666/enVReD2RJCDWNEQTsL4bUMWjbXn+udbwyGFsL/nPX/L837f8874Gc/bB/P8+N/26GhPj7on5VH/KI16ponvFq7yWqiILiy/7baURlGpEcjpE5kX62mVZEi5LHjwAss7O1Tdd2m859P68Gjy7JzCnvPvYPhG7dOmX+9darPcjWagtf9POq1gYHvdFXCxyJUkt6VSJ/Xz6hn11JOpaikKT+d892e3eXuyR5v3jtrmvX7NQeQ0l8BPZGc/Vn0PeKTf3t5Hw9++vVe/pk/Z7P+d/UsV7v9eP6UD+19o64Oe2tWw+f+n3PGbv9fc8+8=</diagram></mxfile>
```
Comment (bfontana) [inline libraries-testing/article-resources/archunit/figure1.drawio:1]: I assume this file is the source for the article figure.  If that's true, it shouldn't be part of the source code.

File: libraries-testing/pom.xml [hunk 2]
```diff
@@ -193,7 +200,7 @@
 
     <properties>
         <asciidoctor.version>1.5.7.1</asciidoctor.version>
```
Comment (bfontana) [inline libraries-testing/pom.xml:202]: This property, and the following related to serenity seems to be particular of your environment. If so please remove them.

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/SmurfsRepository.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import com.baeldung.archunit.smurfs.persistence.domain.Smurf;
+
+import static java.util.stream.Collectors.toList;
+
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/SmurfsRepository.java:16]: There shouldn't be Javadoc comments.

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java:1]: Same here. Delete javadoc.

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * 
+ */
+package com.baeldung.archunit.smurfs.persistence.domain;
+
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java:6]: Same here

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    public Smurf() {}
+
+    public Smurf(String name, boolean comic, boolean cartoon) {
+        this.name = name;
+        this.comic = comic;
+        this.cartoon = cartoon;
+    }
+
+    /**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java:23]: Please remove all javadoc comments.

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/SmurfsService.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/SmurfsService.java:1]: Remove javadoc

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/SmurfsService.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Component;
+
+import com.baeldung.archunit.smurfs.persistence.SmurfsRepository;
+import com.baeldung.archunit.smurfs.persistence.domain.Smurf;
+import com.baeldung.archunit.smurfs.service.dto.SmurfDTO;
+
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/SmurfsService.java:15]: Please remove all javadocs.

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/SmurfsRepository.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        smurfs.put("Actor", new Smurf("Actor", true, true));
+        smurfs.put("Alchemist", new Smurf("Alchemist", true, true));
+        smurfs.put("Archeologist", new Smurf("Archeologist", true, true));
+        smurfs.put("Architect", new Smurf("Architect", true, true));
+        smurfs.put("Baby", new Smurf("Baby", true, true));
+        smurfs.put("Baker", new Smurf("Baker", true, true));
+        smurfs.put("Baker", new Smurf("Baker", true, true));
+    }
+    
+    public SmurfsRepository() {
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/SmurfsRepository.java:37]: An empty constructor isn't necessary.

File: libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+ * @author Philippe
+ *
+ */
+public class SmurfsArchUnitTest {
+    
+    @Test
+    public void givenPresentationLayerClasses_thenWrongCheckFails() {        
+        JavaClasses jc = new ClassFileImporter().importPackages("com.baeldung.archunit.smurfs");
+        
+        ArchRule r1 = classes()
```
Comment (bfontana) [inline libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java:30]: There are some issues with Fluent API formatting. The formatted recommended in the Author's documentation should help. Sometimes even after using the formatter, may still be some issues you have to tweak manually.

File: libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+        assertThrows(AssertionError.class, ()-> r1.check(jc)) ;
+    }
+    
+
+    @Test
+    public void givenPresentationLayerClasses_thenCheckWithFrameworkDependenciesSuccess() {        
+        JavaClasses jc = new ClassFileImporter().importPackages("com.baeldung.archunit.smurfs");
+        
+        ArchRule r1 = classes()
+          .that()
```
Comment (bfontana) [inline libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java:46]: Same here

File: libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+        r1.check(jc);
+    }
+
+    @Test
+    public void givenPresentationLayerClasses_thenNoPersistenceLayerAccess() {        
+        JavaClasses jc = new ClassFileImporter().importPackages("com.baeldung.archunit.smurfs");
+        
+        ArchRule r1 = noClasses()
+          .that()
```
Comment (bfontana) [inline libraries-testing/src/test/java/com/baeldung/archunit/smurfs/SmurfsArchUnitTest.java:60]: And here

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
+/**
```
Comment (bfontana) [inline libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java:1]: Remove javadoc comments

File: libraries-testing/pom.xml (hunk 2 above)
Comment (psevestre) [reply libraries-testing/pom.xml:202]: In this case, all I did was to fix the identation, which was using tabs, to the recommended number of spaces. All other properties were already there - except, of course, the ArchUnit's

**Comment by psevestre** [comment]
Hi @bfontana : I've made the fixes as requested (except for the pom.xml change - see comment on that one)

File: libraries-testing/pom.xml (hunk 2 above)
Comment (bfontana) [reply libraries-testing/pom.xml:202]: Ok. Thanks

File: libraries-testing/article-resources/archunit/figure1.drawio (hunk 1 above)
Comment (psevestre) [reply libraries-testing/article-resources/archunit/figure1.drawio]: Removed

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java (hunk 13 above)
Comment (psevestre) [reply libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java]: Removed

Comment (psevestre) [reply libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java]: Removed

Comment (psevestre) [reply libraries-testing/src/main/java/com/baeldung/archunit/smurfs/service/dto/SmurfDTO.java]: Removed

File: libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java (hunk 6 above)
Comment (psevestre) [reply libraries-testing/src/main/java/com/baeldung/archunit/smurfs/persistence/domain/Smurf.java]: Removed

**Pull request merged**